pkg/controller/component/synthesize_component.go

```diff
@@ -39,9 +39,10 @@ import (
 // BuildSynthesizedComponent builds a new SynthesizedComponent object, which is a mixture of component-related configs from ComponentDefinition and Component.
 func BuildSynthesizedComponent(reqCtx intctrlutil.RequestCtx,
 	cli client.Reader,
+	cluster *appsv1alpha1.Cluster,
```

Comment:
this function is intended not to depend on the cluster object anymore, where is it needed now?

Reply:
if the cluster is nil,  Comp2CompDefs of the SynthesizedComponent will be nil. then, I can not ref other componentDef service/account to `vars`